spring-native-configuration/src/main/java/org/springframework/integration/IntegrationHints.java

```diff
@@ -0,0 +1,123 @@
+
+import org.springframework.nativex.hint.AccessBits;
+import org.springframework.nativex.hint.InitializationHint;
+import org.springframework.nativex.hint.InitializationTime;
+import org.springframework.nativex.hint.NativeHint;
+import org.springframework.nativex.hint.ProxyHint;
+import org.springframework.nativex.hint.ResourceHint;
+import org.springframework.nativex.hint.TypeHint;
+import org.springframework.nativex.type.NativeConfiguration;
+
+@NativeHint(trigger = org.springframework.integration.config.EnableIntegration.class,
+		initialization =
+		@InitializationHint(initTime = InitializationTime.BUILD,
+				types = {
+						org.springframework.integration.config.IntegrationRegistrar.class,
+						org.springframework.integration.support.json.JacksonPresent.class,
+						org.springframework.integration.support.management.micrometer.MicrometerMetricsCaptorRegistrar.class,
+						org.springframework.integration.http.config.HttpContextUtils.class,
+						org.springframework.integration.websocket.config.WebSocketIntegrationConfigurationInitializer.class
+				}),
+		resources = @ResourceHint(patterns = "META-INF/spring.integration.properties"),
+		types = {
+				@TypeHint(access = AccessBits.FULL_REFLECTION,
+						types = {
+								org.springframework.integration.gateway.RequestReplyExchanger.class,
+								org.springframework.integration.graph.Graph.class,
+								org.springframework.integration.graph.CompositeMessageHandlerNode.class,
+								org.springframework.integration.graph.DiscardingMessageHandlerNode.class,
+								org.springframework.integration.graph.EndpointNode.class,
+								org.springframework.integration.graph.ErrorCapableCompositeMessageHandlerNode.class,
+								org.springframework.integration.graph.ErrorCapableDiscardingMessageHandlerNode.class,
+								org.springframework.integration.graph.ErrorCapableEndpointNode.class,
+								org.springframework.integration.graph.ErrorCapableMessageHandlerNode.class,
+								org.springframework.integration.graph.ErrorCapableRoutingNode.class,
+								org.springframework.integration.graph.IntegrationNode.class,
+								org.springframework.integration.graph.LinkNode.class,
+								org.springframework.integration.graph.MessageChannelNode.class,
+								org.springframework.integration.graph.MessageGatewayNode.class,
+								org.springframework.integration.graph.MessageHandlerNode.class,
+								org.springframework.integration.graph.MessageProducerNode.class,
+								org.springframework.integration.graph.MessageSourceNode.class,
+								org.springframework.integration.graph.PollableChannelNode.class,
+								org.springframework.integration.graph.ReceiveCounters.class,
+								org.springframework.integration.graph.RoutingMessageHandlerNode.class,
+								org.springframework.integration.graph.SendTimers.class,
+								org.springframework.integration.graph.TimerStats.class,
+								org.springframework.integration.http.management.IntegrationGraphController.class,
```

Comment:
I assumed this one should be present in the reflection info according to `WebComponentProcessor`, but it is not visible somehow.
It has a `@RestController` on it and is registered as a bean functional way in the `IntegrationGraphControllerRegistrar`, so not sure what is going on here.

Reply:
OK I will have a deeper look.